### ISSUE TITLE
[Feature Request] Add support for QD format

### ISSUE THREAD

**Issue by Anuskuss** [SourMesen/Mesen#625]
No-Intro added some new [FDS ROMs](https://datomatic.no-intro.org/index.php?page=show_record&s=31&n=0333) but they have a .qd instead of the usual .fds extension. I did some research and it seems like the difference between those two formats isn't huge and I think the reason why it doesn't work right now is because of the checksum or something? Anyway, if support isn't possible natively (maybe the BIOS won't let it start) it would be neat if there was some kind of soft-patching happening in the background so the ROM starts atleast.

See this script for information about the format:
https://gist.github.com/einstein95/6545066905680466cdf200c4cc8ca4f0

**Comment by SourMesen** [comment]
Thanks for the request.

.qd files have been around for a long time, I believe they are what Nintendo used in its own emulators.
I'm not really convinced adding more fragmentation to the file formats used by NES emulators is really helping anybody, though.

Unless the format gets supported by other popular NES emulators, I'd tend to err on the side of caution and not add support for it yet.

**Comment by Anuskuss** [comment]
Fair enough, it's your choice - but if it's an official Nintendo product, it has to be emulated IMO.

**Comment by SourMesen** [comment]
This isn't so much another "product" as it is another representation of the same data, though.  This is a bit like another request that requested support for MAME's split rom format for NES games.  The more formats emulators add over time, the more complex things become for the end users (especially when you start thinking about rom hacks distributed as IPS patches, etc.).  

I added support for header-less .nes roms a couple of years ago based on a feature request like this, and in hindsight, it was a bad move.  Headerless NES roms make no sense and do nothing other than force emulator authors to use a game database to be able to load the games, and it makes it impossible to use rom hacks, etc.  I'm trying to avoid making another similar mistake, which is why I'd prefer to see if other emulators add support for this first.

**Comment by 20Phoenix** [comment]
Yeah, QD files support would be could.
NES Online on Switch uses some qd files aswell.
I would love to be able to check them with Mesen 😄
